packages/venia-concept/src/components/Navigation/navigation.js

```diff
@@ -26,46 +31,136 @@ const tiles = CATEGORIES.map(category => (
+                <SignIn showCreateAccountForm={this.showCreateAccountForm} />
+            </div>
+        );
+    }
+
+    get createAccountForm() {
+        const { classes } = this.props;
+        const className =
+            !this.state.isCreateAccountOpen || this.props.isSignedIn
+                ? classes.createAccountClosed
+                : classes.createAccountOpen;
+        return (
+            <div className={`${className} ${classes.signInForm}`}>
+                <NavHeader
+                    onBack={this.hideCreateAccountForm}
+                    title={'Create Account'}
+                />
+                <CreateAccount />
+            </div>
+        );
+    }
+
+    showSignInForm = () => {
+        this.setState({
+            isSignInOpen: true
+        });
+    };
+
+    hideSignInForm = () => {
+        this.setState({
+            isSignInOpen: false
+        });
+    };
+
+    showCreateAccountForm = () => {
+        this.setState({
+            isCreateAccountOpen: true
+        });
+    };
+
+    hideCreateAccountForm = () => {
+        this.setState({
+            isCreateAccountOpen: false
+        });
     };
 
     render() {
         const { classes, isOpen } = this.props;
         const className = isOpen ? classes.open : classes.closed;
+        const { bottomDrawer, signInForm, createAccountForm } = this;
 
         return (
             <aside className={className}>
-                <div className={classes.header}>
-                    <h2 className={classes.title}>
-                        <span>Main Menu</span>
-                    </h2>
-                    <Trigger>
-                        <Icon name="x" />
-                    </Trigger>
-                </div>
+                <NavHeader title={'Main Menu'} />
                 <nav className={classes.tiles}>{tiles}</nav>
-                <ul className={classes.items}>
-                    <li className={classes.item}>
-                        <span className={classes.link}>
-                            <Icon name="user" />
-                        </span>
-                    </li>
-                    <li className={classes.item}>
-                        <span className={classes.link}>
-                            <Icon name="heart" />
-                        </span>
-                    </li>
-                    <li className={classes.item}>
-                        <span className={classes.link}>
-                            <Icon name="map-pin" />
-                        </span>
-                    </li>
-                </ul>
+                <div className={classes.bottomDrawer}>{bottomDrawer}</div>
+                {signInForm}
+                {createAccountForm}
             </aside>
         );
     }
 }
 
-export default classify(defaultClasses)(Navigation);
+const mapStateToProps = state => {
+    return {
+        isSignedIn: state['user']['isSignedIn'],
```

Comment:
I'm surprised the linter didn't turn these into dot-lookup props here. Why are they bracket lookups?

Reply:
Whoops, just a carry over from programming too much in python. We will change it to dot lookup. :+1: